src/content/blog/astro-490.mdx

```diff
@@ -0,0 +1,137 @@
+	const result = await container.renderToString(Card, {
+		slots: {
+			default: 'Card content',
+		},
+	});
+
+	expect(result).toContain('This is a card');
+	expect(result).toContain('Card content');
+});
+```
+
+To get started with testing Astro components with Vitest, we made [a new example](https://github.com/withastro/astro/tree/main/examples/container-with-vitest) you can checkout, or create a new project with directly:
+
+```shell
+npm create astro@latest -- --template container-with-vitest
+```
+
+[Our testing guide](https://docs.astro.build/en/guides/testing/#vitest-and-container-api) has also been updated with more information on how to use the Container API for testing. Test everything!
+
+To learn more about the Container API and how to use it, check out the [Container API documentation](https://docs.astro.build/en/reference/container-reference/). To give feedback on this feature, visit the [Container API RFC](https://github.com/withastro/roadmap/pull/916).
+
+## Stabilized experimental features
+
+This release stabilize the CRSF protection (introduced in Astro 4.6) and i18n domains support (introduced in Astro 4.3) features. As such, it is no longer necessary to enable the `experimental.csrfProtection` and `experimental.i18nDomains` options in your Astro config to use these features.
+
+The `experimental.security.csrfProtection` option has been removed in favor of a new `security.checkOrigin` option. To enable CRSF protection, set `security.checkOrigin` to `true` in your Astro config and remove the `experimental.security.csrfProtection` option if you have it enabled:
+
+```diff
+import { defineConfig } from "astro/config";
+
+export default defineConfig({
++	security: {
++    checkOrigin: true
++  }
+-	experimental: {
+-    security: {
+-      csrfProtection: {
+-        origin: true
+-      }
+-    }
+-	},
+});
+```
+
+As for i18n domains, you can keep your configuration as is, just without the `experimental.i18nDomains` option:
+
+```diff
+import { defineConfig } from "astro/config";
+
+export default defineConfig({
+  site: "https://example.com",
+  output: "server",
+  adapter: node({
+    mode: 'standalone',
+  }),
+  i18n: {
+    defaultLocale: "en",
+    locales: ["en", "fr", "pt-br", "es"],
+    prefixDefaultLocale: false,
+    domains: {
+      fr: "https://fr.example.com",
+      es: "https://example.es",
+    },
+  },
+-  experimental: {
+-    i18nDomains: true,
+-  },
+});
+```
+
+For more information on these features, see the [security](https://docs.astro.build/en/reference/configuration-reference/#security) and [i18n domains](https://docs.astro.build/en/guides/internationalization/#domains) sections of the Astro documentation respectively.
+
+## Bug Fixes
+
+As it do be, Astro 4.9 includes more bug fixes and smaller improvements that couldn't make it into this post! Check out the full [release notes](https://github.com/withastro/astro/blob/refs/heads/main/packages/astro/CHANGELOG.md#490) to learn more!
```

Comment:
I simply cannot change this.

Reply:
It is very important that people know there's a human changing the text there on every update.